store/details_v2.go

```diff
@@ -87,6 +87,139 @@ type storeSnapMedia struct {
 	Height int64  `json:"height"`
 }
 
+// storeInfoChannel is the channel description included in info results
+type storeInfoChannel struct {
+	Architecture string `json:"architecture"`
+	Name         string `json:"name"`
+	Risk         string `json:"risk"`
+	Track        string `json:"track"`
+}
+
+// storeInfoChannelSnap is the snap-in-a-channel of which the channel map is made
+type storeInfoChannelSnap struct {
+	storeSnap
+	Channel storeInfoChannel `json:"channel"`
+}
+
+// storeInfo is the result of v2/info calls
+type storeInfo struct {
+	ChannelMap []*storeInfoChannelSnap `json:"channel-map"`
+	Snap       storeSnap               `json:"snap"`
+	Name       string                  `json:"name"`
+	SnapID     string                  `json:"snap-id"`
+}
+
+func infoFromStoreInfo(si *storeInfo) (*snap.Info, error) {
+	if len(si.ChannelMap) == 0 {
+		// if a snap has no released revisions, it _could_ be returned
+		// (currently no, but spec is purposely ambiguous)
+		// we treat it as a 'not found' for now at least
+		return nil, ErrSnapNotFound
+	}
+
+	thisOne := si.ChannelMap[0]
+	// here we assume that the ChannelSnapInfo can be populated with data
+	// that's in the channel map and not the outer snap. This is a
+	// reasonable assumption today, but copyNonZeroFrom can easily be
+	// changed to copy to a list if needed.
+	copyNonZeroFrom(&si.Snap, &thisOne.storeSnap)
+
+	info, err := infoFromStoreSnap(&thisOne.storeSnap)
+	if err != nil {
+		return nil, err
+	}
+	info.Channel = thisOne.Channel.Name
+	info.Channels = make(map[string]*snap.ChannelSnapInfo, len(si.ChannelMap))
+	seen := make(map[string]bool, len(si.ChannelMap))
+	for _, s := range si.ChannelMap {
+		ch := s.Channel
+		info.Channels[ch.Track+"/"+ch.Risk] = &snap.ChannelSnapInfo{
+			Revision:    snap.R(s.Revision),
+			Confinement: snap.ConfinementType(s.Confinement),
+			Version:     s.Version,
+			Channel:     ch.Name,
+			Epoch:       s.Epoch,
+			Size:        s.Download.Size,
+		}
+		if !seen[ch.Track] {
+			seen[ch.Track] = true
+			info.Tracks = append(info.Tracks, ch.Track)
```

Comment:
this is a preexisting issue, but I don't see a test about the value of Tracks  (should we test also with more than one track? )

TestInfoAndChannels

doesn't seem very precise, for example doesn't check the length of Channels afaict